File: 1-two-sum/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> dic;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (dic.find(complement) != dic.end()) {
                return {dic[complement], i};
            }
            else {
                dic[nums[i]] = i;
            }
        }   
        return {};
    }
};

File: 1-two-sum/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i in range(len(nums)):
            compl = target - nums[i]
            if compl in dic:
                return [i,dic[compl]]

            dic[nums[i]] = i